### DISCUSSION TITLE
nftables sets 支持

### DISCUSSION THREAD

**Discussion by zfl9** [zfl9/chinadns-ng#114] [Ideas]
由于 ipset 在很多新的发行版被标记为 deprecated，因此打算抽空实现下 nft set 查询接口。

目前想法：增加 nft set 支持，同时保留原有 ipset 支持（且默认 ipset）；如果想使用 nft set，则通过一个命令行选项来切换。

**Comment by zfl9** [comment]
`2023.04.01`版本已加入nftset支持。

**Reply by YaoFei509** [reply]
nft set 如何在规则里引用其他family 其他table 的 set ?
比如 ip saddr @ 怎么写？

**Reply by zfl9** [reply]
https://unix.stackexchange.com/questions/329971/nftables-ip-set-multiple-tables

**Reply by zfl9** [reply]
简单来说：做不到。

**Reply by YaoFei509** [reply]
遇到困难绕着走，nftables做不到没关系，反正-a/-A 选项里可以随便指定自己的set名，那咱既然在inet/nat 表里用这个集合，那就把集合建在这个table里即可。搞定。

`
	-a inet@nat@chnip4,inet@nat@chnip6   -A inet@nat@gfwip4,inet@nat@gfwip6 
`

然后 /etc/nftables.conf里面写

`
table inet nat {
	set chnip4 {
		type ipv4_addr
		flags interval
	}

	set chnip6 {
		type ipv6_addr
		flags interval
	}

	set gfwip4 {
		type ipv4_addr
		flags interval
	}

	set gfwip6 {
		type ipv6_addr
		flags interval
	}

    chain postrouting {
`

即可。
